conjurapi/client.go

```diff
@@ -308,12 +308,12 @@ func (c *Client) RotateAPIKeyRequest(roleID string) (*http.Request, error) {
 	)
 }
 
-func (c *Client) CheckPermissionRequest(resourceID string, privilege string) (*http.Request, error) {
+func (c *Client) CheckPermissionRequest(resourceID string, roleID string, privilege string) (*http.Request, error) {
 	account, kind, id, err := parseID(resourceID)
 	if err != nil {
 		return nil, err
 	}
-	checkURL := makeRouterURL(c.resourcesURL(account), kind, url.QueryEscape(id)).withFormattedQuery("check=true&privilege=%s", url.QueryEscape(privilege)).String()
+	checkURL := makeRouterURL(c.resourcesURL(account), kind, url.QueryEscape(id)).withFormattedQuery("check=true&role=%s&privilege=%s", url.QueryEscape(roleID), url.QueryEscape(privilege)).String()
```

Comment:
I wonder if it's generally an acceptable practice to include a value-less key in a query string, like `?check=true&role=&privilege=read`. The URI RFC doesn't mandate a format for the query string [[1](https://stackoverflow.com/questions/4557387/is-a-url-query-parameter-valid-if-it-has-no-value),[2](https://www.rfc-editor.org/rfc/rfc3986#section-3.4)], and it seems from the tests that Conjur handles the empty value, but maybe we could do some pre-processing to keep the query string clean.